hack/make-release-notes.sh

```diff
@@ -74,7 +74,7 @@ for f in "${download_assets[@]}"; do
 done
 echo
 echo "Thanks to our contributors for helping out with ${TAG}:"
-git log "$(git describe --tags --abbrev=0)..HEAD" --format=%an |
+git log "$(git describe --tags --match 'v*' --abbrev=0 "${TAG}^")..${TAG}" --format=%an |
```

Comment:
we should be checking `prev version..new version` but with what it was it was checking `TAG..HEAD` which is the same commit. not sure what changed in the ci but this should fix it.

some tests i ran locally
```
$ TAG=v0.4.3 bash -c 'git log "$(git describe --tags --match "v*" --abbrev=0 "${TAG}^")..${TAG}" --format=%an | sort | uniq -c | sort -rn | sed -E "s,^(\s+[0-9]+\s),- ,g"'
- Chris Kim
- Ahmet Alp Balkan
- zhengtianbao
- Cornelius Weig
- Alexandre Garnier
- ferhat elmas
- Vincent
- Samuel Walker
- Kazuki Suda
- Kai
- Jun
- Gábor Lipták
- Evgeny Shmarnev
- Chok Yip Lau
- Carlos Alexandro Becker
- Batuhan Apaydın
- Allianzcortex

$ TAG=v0.4.2 bash -c 'git log "$(git describe --tags --match "v*" --abbrev=0 "${TAG}^")..${TAG}" --format=%an | sort | uniq -c | sort -rn | sed -E "s,^(\s+[0-9]+\s),- ,g"'
- Chris Kim
- Cornelius Weig
- Ahmet Alp Balkan
- ferhat elmas
- Vincent
- Evgeny Shmarnev
- Chok Yip Lau
- Alexandre Garnier
```

these match the outputs in the release notes (https://github.com/kubernetes-sigs/krew/releases/tag/v0.4.3 and https://github.com/kubernetes-sigs/krew/releases/tag/v0.4.2)

i added that glob match in case we ever tag with something else (highly unlikely but figured it didnt hurt to add)